### PR TITLE
Improve PyTorch and MXNet ExternalSource examples

### PR DESCRIPTION
**Pull request by JanuszL**
- updates PyTorch and MXNet ExternalSource examples with the most recent recommendations regarding combining DALI iterators and ExternalSource input

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It fixes PyTorch and MXNet ExternalSource examples

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     updates PyTorch and MXNet ExternalSource examples with the most recent recommendations regarding combining DALI iterators and ExternalSource input
 - Affected modules and functionalities:
     MXNet and PyTorch examples
 - Key points relevant for the review:
     wording?
 - Validation and testing:
     CI
 - Documentation (including examples):
     examples has been updated

**JIRA TASK**: *[NA]*

### CODE REVIEW

**Comment by JanuszL** [comment]
!build

File: docs/examples/frameworks/mxnet/mxnet-external_input.ipynb [hunk 1]
```diff
@@ -139,14 +139,26 @@
    "source": [
     "### Using the pipeline\n",
     "\n",
-    "At the end let us see how it works. Please also notice the usage of `last_batch_padded` that tell iterator that the difference between data set size and batch size alignment is padded by real data that could be skipped at when provided to the framework (`fill_last_batch`):"
+    "In the end, let us see how it works.\n",
+    "\n",
+    "`last_batch_padded` and `fill_last_batch` are set here only for the demonstration purpose. The user may write any custom code and change the epoch size epoch to epoch it is recommended to set `size` to -1 and let the iterator just wait for StopIteration exception from the `iter_setup`.\n",
```
Comment (jantonguirao) [inline docs/examples/frameworks/mxnet/mxnet-external_input.ipynb:144]:
```suggestion
    "`last_batch_padded` and `fill_last_batch` are set here only for the demonstration purposes. The user may write any custom code and change the epoch size epoch to epoch. In that case, it is recommended to set `size` to -1 and let the iterator just wait for StopIteration exception from the `iter_setup`.\n",
```
?

File: docs/examples/frameworks/pytorch/pytorch-external_input.ipynb [hunk 2]
```diff
@@ -139,14 +139,26 @@
    "source": [
     "### Using the pipeline\n",
     "\n",
-    "At the end let us see how it works. Please also notice the usage of `last_batch_padded` that tell iterator that the difference between data set size and batch size alignment is padded by real data that could be skipped at when provided to the framework (`fill_last_batch`):"
+    "In the end, let us see how it works.\n",
+    "\n",
```
Comment (jantonguirao) [inline docs/examples/frameworks/pytorch/pytorch-external_input.ipynb:143]: same here

Comment (JanuszL) [reply docs/examples/frameworks/pytorch/pytorch-external_input.ipynb:143]: Done

File: docs/examples/frameworks/mxnet/mxnet-external_input.ipynb (hunk 1 above)
Comment (JanuszL) [reply docs/examples/frameworks/mxnet/mxnet-external_input.ipynb]: Done

**Comment by JanuszL** [comment]
!build

**Pull request merged**